src/graph.cpp

```diff
@@ -186,25 +186,25 @@ inline std::vector<std::vector<node_id_t>> Graph::supernodes_to_merge(std::pair<
   // did end up being able to merge after all
   std::vector<node_id_t> temp_vec;
   for (node_id_t a : new_reps)
-    if (to_merge[a].size() == 0) temp_vec.push_back(a);
+    if (to_merge[a].empty()) temp_vec.push_back(a);
   std::swap(new_reps, temp_vec);
 
   // add to new_reps all the nodes we will merge into
   for (node_id_t a = 0; a < num_nodes; a++)
-    if (to_merge[a].size() != 0) new_reps.push_back(a);
+    if (!to_merge[a].empty()) new_reps.push_back(a);
 
   reps = new_reps;
   return to_merge;
 }
 
 inline void Graph::merge_supernodes(Supernode** copy_supernodes, std::vector<node_id_t> &new_reps,
-               std::vector<std::vector<node_id_t>> &to_merge, bool make_copy, bool first_round) {
+               std::vector<std::vector<node_id_t>> &to_merge, bool first_round, bool make_copy) {
```

Comment:
Nice catch. That might have explained part of the difference in performance we were observing.

Reply:
Actually, semantically it probably doesn't make a difference. I think we can (and should) reduce the number of arguments to this function to just one 'make_copy' and pass in whether we require the supernodes to be backed up. (first_round && make_copy) or just first_round which starts true only if make_copy is true.

Reply:
sure, ill make that change